src/graft_dag.cpp

```diff
@@ -0,0 +1,524 @@
+// Copyright 2019-2022 bito project contributors.
+// bito is free software under the GPLv3; see LICENSE file for details.
+
+#include "graft_dag.hpp"
+
+#include "gp_dag.hpp"
+#include "quartet_hybrid_request.hpp"
+#include "reindexer.hpp"
+#include "subsplit_dag.hpp"
+
+// ** Constructors
+
+GraftDAG::GraftDAG(SubsplitDAG &dag) : host_dag_(dag) {}
+
+// ** Comparators
+
+int GraftDAG::Compare(GraftDAG &dag_a, GraftDAG &dag_b) {
+  SubsplitDAG &host_a = dag_a.GetHostDAG();
+  SubsplitDAG &host_b = dag_b.GetHostDAG();
+  // (1) Compare host DAGs.
+  int host_diff = SubsplitDAG::Compare(host_a, host_b);
+  if (host_diff != 0) {
+    return host_diff;
+  }
+  // (2) Compare graft nodes.
+  BitsetVector nodes_a = dag_a.GetSortedVectorOfNodeBitsets();
+  BitsetVector nodes_b = dag_b.GetSortedVectorOfNodeBitsets();
+  for (size_t i = 0; i < std::min(nodes_a.size(), nodes_b.size()); i++) {
```

Comment:
We do this comparison between BitsetVectors 4 times in this function and the one below.

That definitely merits a subroutine! 

BTW, we couldn't just use normal `<` for these data types? Don't we have < for Bitsets?

Reply:
Fixed using `<`.